src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3309,6 +3336,19 @@ bool StorageReplicatedMergeTree::fetchPart(const String & part_name, const Stora
             if (quorum)
                 updateQuorum(part_name);
 
+            /// alexelex: i'm not sure in witch order should that and merge_selecting_task->schedule() be
+            /// merged parts that are still inserted with quorum. if it only contains one block, it hasn't been merged before
+            if (part_info.max_block != part_info.min_block)
```

Comment:
`part_info.level != 0 ||  part_info.mutation != 0` is better. Because max_block and min_block can be equal, but part was already merged (`optimize final`) or mutated.